src/PKSim.Core/Snapshots/Mappers/SimulationMapper.cs

```diff
@@ -236,14 +235,23 @@ private string[] usedObservedDataFrom(ModelSimulation simulation, PKSimProject p
             .Select(x => x.Name).ToArray();
       }
 
-      private string usedSimulationSubject<T>(ModelSimulation simulation) where T : class, ISimulationSubject
+      private string usedSimulationSubject<T>(ModelSimulation simulation, PKSimProject project) where T : class, ISimulationSubject
       {
-         var name = simulation.BuildingBlock<T>()?.Name;
+         var usedBuildingBlock = simulation.UsedBuildingBlockInSimulation<T>();
+
+         //simulation is not using this building block. 
+         if (usedBuildingBlock == null)
+            return null;
+
+         var templateBuildingBlock = project.BuildingBlockById<T>(usedBuildingBlock.TemplateId);
```

Comment:
@Yuri05 I am now retrieving the templatebuildign block by ID so that it does not matter if sthg was renamed or not. We can always use it. This is what is happening for all other building blocks (Events, Formulation etc... )That's why it was working